### ISSUE TITLE
Problem using typed sortedcontainer with defaultfactory

### ISSUE THREAD

**Issue by ltdt-apex** [h4l/sortedcontainers-stubs#3]
so with dataclass, it is supposed to use default_factory to create stuff, I can't really find a way to make annotation without problems

sample code:
```
from dataclasses import dataclass, field
from sortedcontainers import SortedList

@dataclass
class Example():
    samples: SortedList[str] = field(default_factory=SortedList)
```

error:
```
Argument of type "type[SortedList[_T@SortedList]]" cannot be assigned to parameter "default_factory" of type "() -> _T@field" in function "field"
  No overloaded function matches type "() -> SortedList[str]"
```

**Comment by h4l** [comment]
Thanks for the report. It's probably because of the way `SortedList.__new__` has several overloads but no explicit empty argument overload. I'll see if I can tweak the overloads to make this work.

**Comment by h4l** [comment]
@Yummybatsoup This should be fixed in v2.4.2 which I just released.

**Comment by ltdt-apex** [comment]
Thanks, confirms that it work.
